pontoon/base/static/js/fluent_interface.js

```diff
@@ -6,21 +6,136 @@ var fluentSerializer = new FluentSyntax.FluentSerializer();
 var Pontoon = (function (my) {
 
   /*
-   * Render original string element: simple string value or a variant.
+   * Is ast element of type that can be presented as a simple string:
+   * - TextElement
+   * - Placeable with expression type CallExpression, ExternalArgument
+   *   or MessageReference
    */
-  function renderOriginalElement(value, elements) {
+  function isSimpleElement(element) {
+    if (element.type === 'TextElement') {
+      return true;
+    }
+
+    // Placeable
+    if (
+      element.expression &&
+      [
+        'CallExpression',
+        'ExternalArgument',
+        'MessageReference'
+      ].indexOf(element.expression.type) >= 0
+    ) {
+      return true;
+    }
+
+    return false;
+  }
+
+  /*
+   * Is ast element representing a pluralized string?
+   *
+   * Keys of all variants of such elements are either
+   * CLDR plurals or numbers.
+   */
+  function isPluralElement(element) {
+    if (!(element.expression && element.expression.type === 'SelectExpression')) {
+      return false;
+    }
+
+    var CLDRplurals = ['zero', 'one', 'two', 'few', 'many', 'other'];
+
+    return element.expression.variants.every(function (item) {
+      return (
+        CLDRplurals.indexOf(item.key.name) !== -1 ||
+        item.key.type === 'NumberExpression'
+      );
+    });
+  }
+
+  /*
+   * Is ast of a message, supported in rich FTL editor?
+   *
+   * Message is supported if it's valid and all value elements
+   * and all attribute elements are:
+   * - simple elements or
+   * - select expressions
+   */
+  function isSupportedMessage(ast) {
+    function elementsSupported(elements) {
+      return elements.every(function(element) {
+        return (
+          isSimpleElement(element) ||
+          (element.expression && element.expression.type === 'SelectExpression')
+        );
+      });
+    }
+
+    // Parse error
+    if (ast.type === 'Junk') {
+      return false;
+    }
+
+    var valueSupported = !ast.value || elementsSupported(ast.value.elements);
+    var attributesSupported = ast.attributes.every(function (attribute) {
+      return attribute.value && elementsSupported(attribute.value.elements);
+    });
+
+    return valueSupported && attributesSupported;
+  }
+
+  /*
+   * Is ast of a simple message?
+   *
+   * A simple message has no attributes and all value
+   * elements are simple.
+   */
+  function isSimpleMessage(ast) {
+    if (
+      ast &&
+      !ast.attributes.length &&
+      ast.value &&
+      ast.value.elements.every(function(item) {
+        return isSimpleElement(item);
+      })
+    ) {
+      return true;
+    }
+
+    return false;
+  }
+
+  /*
+   * Render textarea element with the given properties
+   */
+  function renderTextareaElement(id, value, maxlength) {
+    var element = '<textarea class="value" id="ftl-id-' + id + '"';
+
+    if (typeof maxlength !== 'undefined' && maxlength !== null) {
```

Comment:
Is `maxlength = 0` a use case we want to support here? It seems weird to have a textarea in which you can't write anything. To me that should be a disabled textarea instead.

Reply:
No, we shouldn't support `maxlength = 0`. Actually, this condition is always true. I'll remove the `if`.